lib/setupmanagers/qemuhck/qemu_machine.rb

```diff
@@ -212,6 +212,7 @@ def initialize(options)
       @devices_list.flatten!
       @devices_list.compact!
       load_devices
+      @image_path = @sm.boot_device_command(option_config('boot_device'), @run_opts, full_replacement_map).last
```

Comment:
boot_device_command calls create_boot_snapshot, so after you change QemuHCK will generate a snapshot 2 times. This looks bad and can cause unexpected behavior in the future like data loss or vice versa we will have a snapshot but expect that it does not exist.